### PR TITLE
Add drip-swag.

### PR DESCRIPTION
**Pull request by imstar15**
Test room:
https://discord.gg/JX7hqDqb

image

View in explorer:

https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Frpc.testnet.oak.tech#/explorer/query/0x75f8f00dfacbc4d3b466f68696f6c01dd204036ee05a6eb6dafe3ccc75cb010b

image

### CODE REVIEW

**Comment by irsal** [comment]
I tested on the discord room, and wasn't able to see the scheduled tasks on chain.

Screen Shot 2022-04-26 at 11 17 01 AM

Screen Shot 2022-04-26 at 11 17 51 AM

**Comment by irsal** [comment]
I guess we're using the test discord with a different chain. The faucet bot in production has a working `drip-later` command

File: src/bot/actions.js [hunk 1]
```diff
@@ -109,17 +82,43 @@ const dripLater = async (sender, address, time) => {
...
+  
+  const { amount } = dripActions[DRIP_TYPE.LATER];
+  const { data: { hash, providerId } } = res;
 
+  return `I will send ${amount} NEU to address ${address} at ${time} UTC. Extrinsic hash: ${hash}. Your provided_id: ${providerId}.`;
+}
+
+const dripSwag = async (sender, address) => {
+	if (_.isEmpty(address)) {
+    return 'please enter a wallet address after !drip-later.';
```
Comment (irsal) [inline src/bot/actions.js]: `!drip-swag`

File: src/server/actions.js [hunk 2]
```diff
@@ -15,20 +17,50 @@ class Actions {
...
+    const providerId = uuid.v4();
+
+    const now = moment().utc();
+    const executionTimes = [];
+    for (let i = 1; i <= 24; i += 1) {
+      executionTimes.push(Math.ceil(now.add(i, 'hours').valueOf() / 1000));
+    }
+
+    const extrinsic = this.api.tx.automationTime.scheduleNativeTransferTask(providerId, executionTimes, address, amount);
+    const hash = await extrinsic.signAndSend(this.account, { nonce: -1 });
```
Comment (irsal) [inline src/server/actions.js]: @justinzhou93 - this the appropriate way to manage the nonce?

**Review by irsal — CHANGES_REQUESTED** [review]

File: src/server/actions.js (hunk 2 above)
Comment (justinzhou93) [reply src/server/actions.js]: hm idk what a negative nonce means. nonce doesn't really matter unless you're making more than 1 call for a wallet while a call with an existing nonce is still in the transaction pool. so in this case, drip seems to be scheduling 1 task at a time for a given wallet, so therefore, no issue.

Comment (imstar15) [reply src/server/actions.js]: https://polkadot.js.org/docs/api/cookbook/tx#how-do-i-take-the-pending-tx-pool-into-account-in-my-nonce

About "nonce = -1", this is an official example.

_As a convenience function, the accountNextIndex can be omitted by specifying a nonce of -1, allow the API to do the lookup. In this case the above can be simplified even further,_

**Comment by imstar15** [comment]
> I tested on the discord room, and wasn't able to see the scheduled tasks on chain.
> 
> Screen Shot 2022-04-26 at 11 17 01 AM
> 
> Screen Shot 2022-04-26 at 11 17 51 AM

Let me merge the "fix drip lter" PR and redeploy it.

**Review by irsal — APPROVED** [review]

**Pull request merged**